c2corg_ui/static/js/map/map.js

```diff
@@ -423,14 +423,54 @@ app.MapController.prototype.handleDraw_ = function(event) {
  * @private
  */
 app.MapController.prototype.handleModify_ = function(event) {
-  // TODO handle lines as well, not only points
-  if (this.drawType != 'Point') {
-    alert('Feature type not supported for editing');
-    return;
-  }
   var feature = event.features.item(0);
   this.scope_.$root.$emit('mapFeatureChange', feature);
 };
 
 
+/**
+ * @private
+ */
+app.MapController.prototype.addTrackImporter_ = function() {
+  var dragAndDropInteraction = new ol.interaction.DragAndDrop({
+    formatConstructors: [
+      ol.format.GPX
+    ]
+  });
+  dragAndDropInteraction.on('addfeatures', function(event) {
+    var features = event.features;
+    if (features.length) {
+      var source = this.getVectorLayer_().getSource();
+      // TODO: keep associated features?
+      source.clear();
+      var feature = this.processFeature_(features[0]);
+      source.addFeature(feature);
+      this.map.getView().fit(
+          source.getExtent(), /** @type {ol.Size} */ (this.map.getSize()));
+      this.scope_.$root.$emit('mapFeatureChange', feature);
+    }
+  }.bind(this));
+  this.map.addInteraction(dragAndDropInteraction);
+};
+
+
+/**
+ * @param {ol.Feature} feature Feature to process.
+ * @return {ol.Feature}
+ * @private
+ */
+app.MapController.prototype.processFeature_ = function(feature) {
+  var geometry = feature.getGeometry();
+  // convert MultiLineString to LineString
```

Comment:
Indeed. For some reason I thought the API only accepted ``LineString`` but I have just checked and ``MultiLineString``. Then no reason to make such a conversion, I have dropped it. Thanks.